bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java

```diff
@@ -296,10 +296,16 @@ public void onChanged(final List<CidrAddress> added, final List<CidrAddress> rem
         removed.forEach(i -> {
             logger.trace("removed interface {}", i.getAddress().toString());
             if (i.getAddress().equals(networkInterface)) {
-                this.bridge.stop();
-                this.homekitServer.stop();
-                this.homekitServer = null;
-                this.bridge = null;
+                final @Nullable HomekitRoot bridge = this.bridge;
+                if (this.bridge != null) {
+                    this.bridge.stop();
+                    this.bridge = null;
+                }
+                final @Nullable HomekitServer homekitServer = this.homekitServer;
+                if (homekitServer != null) {
+                    this.homekitServer.stop();
+                    this.homekitServer = null;
+                }
```

Comment:
The proper way to avoid compilation warnings is:
```
                final @Nullable HomekitRoot bridge = this.bridge;
                if (bridge != null) {
                    bridge.stop();
                    this.bridge = null;
                }
                final @Nullable HomekitServer homekitServer = this.homekitServer;
                if (homekitServer != null) {
                    homekitServer.stop();
                    this.homekitServer = null;
                }
```

Reply:
right, make sense. this.bridge could have become not null via other thread. but local bridge is final. 
the best - i copied the correct version from another place and "improved" it. 
thank you for pointing out. fixed.

Reply:
I would have preferred i`f (bridge != null) {` rather than `if (this.bridge != null) {` but I hope it is ok like it is.

Reply:
@lolodomo 
argh.. i checked 3 times and asked myself what else i could have missed :(  completely blind. 
3 error in 4 lines :(
need to learn to pay more attention to details. 
thank for your patience 
will take with the next feature-PR.